plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -458,8 +459,13 @@ private DeltaLakeTransactionLogEntry buildAddEntry(ConnectorSession session, Blo
         log.debug("Block %s has %s fields", block, addEntryRow.getFieldCount());
         CheckpointFieldReader add = new CheckpointFieldReader(session, addEntryRow, type);
 
-        String path = add.getString("path");
         Map<String, String> partitionValues = add.getMap(stringMap, "partitionValues");
+        Map<String, Optional<String>> canonicalPartitionValues = canonicalizePartitionValues(partitionValues);
+        if (!(partitionConstraint.isAll() || partitionMatchesPredicate(canonicalPartitionValues, partitionConstraint.getDomains().orElseThrow()))) {
```

Comment:
nit: would be a bit easier to read as `!ipartitionConstraintsAll() && !partitionMatchesPredicate...`
Consider updating `partitionMatchesPredicate` to take `partitionConstraint` as input and do the `isAll` check internally.

Reply:
> ...

I tried introducing a preparatory commit with this method and noticed that 3 of the usages of the existing `partitionMatchesPredicate` method do not fully fit to make use of the newly method (duplicated checks, `partitionContraint` contains additional filtering). I'm refraining now from doing the suggested refactoring.

```
    public static boolean partitionMatchesPredicate(Map<String, Optional<String>> partitionKeys, TupleDomain<DeltaLakeColumnHandle> partitionConstraint)
    {
        if (partitionConstraint.isAll()) {
            return true;
        }

        Map<DeltaLakeColumnHandle, Domain> domains = partitionConstraint.getDomains().orElseThrow();
        return partitionMatchesPredicate(partitionKeys, domains);
    }
```